armory/art_experimental/poison_detection/random_filter.py

```diff
@@ -0,0 +1,49 @@
+import numpy as np
+
+from art.defences.detector.poison import PoisonFilteringDefence
+
+from armory.logs import log
+
+
+class RandomFilterBaselineDefense(PoisonFilteringDefence):
+    def __init__(self, classifier, x_train, y_train, **kwargs):
+        """
+        Create a :class:`.RandomFilterBaselineDefense` object with the provided classifier.
+        :param classifier: Model evaluated for poison.
+        :param x_train: dataset used to train the classifier.
+        :param y_train: labels used to train the classifier.
+        """
+        super().__init__(classifier, x_train, y_train)
+        self.n_data = len(y_train)
+
+    def evaluate_defence(self, is_clean, **kwargs):
```

Comment:
I'm lacking context on `evaluate_defence()`...is this a part of the ART API that all poisoning defenses must have this method?

If no, should we just omit it? I like raising a NotImplementedError if we expect someone might try using the method, but it's not actually important for our sake to implement it

Reply:
It's a `@abc.abstractmethod` in ART, so I think we need to implement it. The NotImplementedError seems right.

Reply:
Got it. In that case, I agree the NotImplementedError makes sense

As an aside, not sure why it didn't err then when I ran the config after deleting the method

Reply:
It did throw some kind of error for me before I implemented it